Source/WebKit/UIProcess/API/Cocoa/WKDownload.mm

```diff
@@ -131,20 +131,10 @@ void decideDestinationWithSuggestedFilename(WebKit::DownloadProxy& download, con
             }
 
             NSFileManager *fileManager = [NSFileManager defaultManager];
-            if (![fileManager fileExistsAtPath:[destination URLByDeletingLastPathComponent].path]) {
-                RunLoop::main().dispatch([download] {
-                    download->didFail([NSError errorWithDomain:NSURLErrorDomain code:NSURLErrorCannotCreateFile userInfo:nil], { });
-                });
-                completionHandler(WebKit::AllowOverwrite::No, { });
-                return;
-            }
-            if ([fileManager fileExistsAtPath:destination.path]) {
-                RunLoop::main().dispatch([download] {
-                    download->didFail([NSError errorWithDomain:NSURLErrorDomain code:NSURLErrorCannotCreateFile userInfo:nil], { });
-                });
-                completionHandler(WebKit::AllowOverwrite::No, { });
-                return;
-            }
+            if (![fileManager fileExistsAtPath:[destination URLByDeletingLastPathComponent].path])
+                return completionHandler(WebKit::AllowOverwrite::No, { });
```

Comment:
is there a way of letting the client know what went wrong? because if they give a URL that they think is valid but is actually invalid, it will seem as if it got cancelled for no reason when in actuality it was cancelled because of one of these reasons. Maybe a release log or something?

Reply:
oh nvm, they are completely redundant